### PR TITLE
Remove redundant Aggregate when `DISTINCT` & `GROUP BY` are in the same query

### PR DESCRIPTION
**Pull request by mertak-synnada**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Inefficient planning is produced when a redundant DISTINCT & GROUP BY clause is used together.

Eg.

```
EXPLAIN SELECT c3 FROM aggregate_test_100 GROUP BY c3 LIMIT 5;
logical_plan
Limit: skip=0, fetch=5
--Aggregate: groupBy=[[aggregate_test_100.c3]], aggr=[[]]
----Aggregate: groupBy=[[aggregate_test_100.c3]], aggr=[[]]
------TableScan: aggregate_test_100 projection=[c3]
physical_plan
GlobalLimitExec: skip=0, fetch=5
--AggregateExec: mode=Final, gby=[c3@0 as c3], aggr=[], lim=[5]
----CoalescePartitionsExec
------AggregateExec: mode=Partial, gby=[c3@0 as c3], aggr=[], lim=[5]
--------RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1
----------AggregateExec: mode=Final, gby=[c3@0 as c3], aggr=[], lim=[5]
------------CoalescePartitionsExec
--------------AggregateExec: mode=Partial, gby=[c3@0 as c3], aggr=[], lim=[5]
----------------RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1
------------------CsvExec: file_groups={1 group: [[WORKSPACE_ROOT/testing/data/csv/aggregate_test_100.csv]]}, projection=[c3], has_header=true
```

Consecutive empty aggregates with same group by are unnecessary.

## What changes are included in this PR?

It's fixed that if a distinct is the same as a following group by statement, it should not be turned into an Aggregate plan.

## Are these changes tested?

Yes

## Are there any user-facing changes?

No

### CODE REVIEW

**Review by ozankabak — APPROVED** [review]
I reviewed this and it LGTM. Easy win!

File: datafusion/sqllogictest/test_files/aggregate.slt [hunk 1]
```diff
@@ -4536,19 +4536,14 @@ EXPLAIN SELECT DISTINCT c3 FROM aggregate_test_100 group by c3 limit 5;
 logical_plan
 01)Limit: skip=0, fetch=5
 02)--Aggregate: groupBy=[[aggregate_test_100.c3]], aggr=[[]]
-03)----Aggregate: groupBy=[[aggregate_test_100.c3]], aggr=[[]]
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/aggregate.slt:4539]: These certainly look like bettr plans to me. 

However, I wonder if these changes now mean the query doesn't fulfill the original intent of the tests 

```sql
#
# Push limit into distinct group-by aggregation tests
#
```

Therefore I wonder if we should  add another test like

```
EXPLAIN SELECT DISTINCT c3, min(c1) FROM aggregate_test_100 group by c3 limit 5;
```

**Review by alamb — APPROVED** [review]
Thank you @mertak-synnada  and @ozankabak

File: datafusion/sqllogictest/test_files/aggregate.slt (hunk 1 above)
Comment (ozankabak) [reply datafusion/sqllogictest/test_files/aggregate.slt:4539]: Makes sense

Comment (alamb) [reply datafusion/sqllogictest/test_files/aggregate.slt:4539]: FWIW in my opinion it would be fine to add the other tests as a new PR too

Comment (ozankabak) [reply datafusion/sqllogictest/test_files/aggregate.slt:4539]: Sounds good, let's do it that way

**Pull request merged**